junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/DefaultLauncherConfig.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.launcher.core;
+
+import static java.util.Collections.unmodifiableCollection;
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.util.Collection;
+
+import org.apiguardian.api.API;
+import org.junit.platform.engine.TestEngine;
+import org.junit.platform.launcher.TestExecutionListener;
+
+@API(status = EXPERIMENTAL, since = "1.3.0")
+class DefaultLauncherConfig implements LauncherConfig {
+
+	private final boolean testEngineAutoRegistrationEnabled;
+
+	private final Collection<TestEngine> additionalTestEngines;
+
+	private final boolean testExecutionListenerAutoRegistrationEnabled;
+
+	private final Collection<TestExecutionListener> additionalTestExecutionListeners;
+
+	DefaultLauncherConfig(final boolean testEngineAutoRegistrationEnabled,
```

Comment:
We don't declare constructor parameters as `final` unless necessary.

Please remove the `final` declarations.